examples/flutter_gallery/lib/demo/expansion_panels_demo.dart

```diff
@@ -221,54 +224,57 @@ class _ExpansionPanelsDemoState extends State<ExpasionPanelsDemo> {
             });
           }
 
-          void changeLocation(_Location newLocation) {
-            setState(() {
-              item.value = newLocation;
-            });
-          }
-
-          return new CollapsibleBody(
-            child: new Column(
-              mainAxisSize: MainAxisSize.min,
-              crossAxisAlignment: CrossAxisAlignment.start,
-              children: <Widget>[
-                new Row(
-                  mainAxisSize: MainAxisSize.min,
-                  children: <Widget>[
-                    new Radio<_Location>(
-                      value: _Location.Bahamas,
-                      groupValue: item.value,
-                      onChanged: changeLocation
-                    ),
-                    new Text('Bahamas')
-                  ]
-                ),
-                new Row(
-                  mainAxisSize: MainAxisSize.min,
-                  children: <Widget>[
-                    new Radio<_Location>(
-                      value: _Location.Barbados,
-                      groupValue: item.value,
-                      onChanged: changeLocation
-                    ),
-                    new Text('Barbados')
-                  ]
-                ),
-                new Row(
-                  mainAxisSize: MainAxisSize.min,
-                  children: <Widget>[
-                    new Radio<_Location>(
-                      value: _Location.Bermuda,
-                      groupValue: item.value,
-                      onChanged: changeLocation
-                    ),
-                    new Text('Bermuda')
-                  ]
-                )
-              ]
+          return new Form(
+            key: form2Key,
+            child: new CollapsibleBody(
+              onSave: () { form2Key.currentState.save(); close(); },
+              onCancel: () { form2Key.currentState.reset(); close(); },
+              child: new FormField<_Location>(
```

Comment:
are you imagining that eventually we'll have a RadioButtonGroupField or something which generates all the radio buttons and so forth, similar to how we do InputFormField for an Input?